UMM-G support (ER)

esr0004 opened the pull request:
Added UMM-G support for the following checks:

- Bounding Coordinate Logic Check
- Vertical Spatial Domain Type Check
- Depth Unit Controlled Vocabulary Check
- Altitude Unit Controlled Vocabulary Check
- Temporal Extent Requirement Check
- Default Date Check

svbagwell reviewed (approved):
LGTM

spa0002 reviewed (commented):
Attaching file with potential edits:
- new rule for umm-g schema due to different  "data" for controlled_keywords_check in other schemas
- Updates to depth_unit_check, altitude_unit_check based on rule mapping
- [altitude_depth_units_ummg.txt](https://github.com/NASA-IMPACT/pyQuARC/files/8490140/altitude_depth_units_ummg.txt)

esr0004 commented:
@jenny-m-wood @spa0002 I changed the Vertical Spatial Domain Type Check, Depth Unit Controlled Vocabulary Check, and Altitude Unit Controlled Vocabulary Check back to their original state for UMM-C and created separate checks for UMM-G (Vertical Spatial Domain Unit Check UMM-G and Vertical Spatial Domain Type Check UMM-G) in the rule_mapping.json file. Also updated the check_messages.json for these checks.
